### ISSUE TITLE
Allow using an external validation set in the SequentialFeatureSelector

### ISSUE THREAD

**Issue by rasbt** [rasbt/mlxtend#436]
Currently, it's unfortunately not possible to use your own validation dataset with the `SequentialFeatureSelector`. However, it would be nice to support to support this. 

For instance one could extend the API to accept a validation set as argument for "cv" like

`SequentialFeatureSelector(cv=own_valididation_set)` where `own_valididation_set` is a dictionary with 

    own_valididation_set['features'] = X_valid
    own_valididation_set['targets'] = y_valid

**Comment by kota7** [comment]
Can we use [PredefinedSplit](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.PredefinedSplit.html) to specify own validation set?

If I read the source correct, `cv` is used only with `cross_val_score` from scikit-learn.  Since `cross_val_score` accepts cross-validation generators as `cv`, `SequentialFeatureSelector` may work with them too (I have not tested yet).

**Comment by rasbt** [comment]
Oh nice, I didn't know that this class existed. I was actually thinking about implementing sth like that, but this definitely solves the problem now. Will add an example to the docs.
